### PR TITLE
Update for 0.7.1

### PR DESCRIPTION
**Pull request by briancurtin**
This change includes the following changes to work with the new OpenStack SDK v0.7.1

* Connection now uses rackspaceauth, as the upstream SDK now depends on keystoneauth.
* service_filter was relocated out of openstack.auth
* The addition of _find to BaseProxy means the find implementations in the database and backup proxy classes is slightly simplified, and falls in line with how every other proxy method is called.

### CODE REVIEW

**Comment by briancurtin** [comment]
@WalterLenuzza can you have a look at this, at least around the database and backup things you did? We added a few things in OpenStack SDK that made the find calls in proxies a little more clean, and there's a change in database/database_service.py around a service_type argument that is now required due to some under-the-hood stuff we get out of keystoneauth. I did a quick check against public cloud and it appeared fine.

File: rackspace/database/database_service.py [hunk 1]
```diff
@@ -21,5 +21,6 @@ class DatabaseService(service_filter.ServiceFilter):
     def __init__(self, version=None):
         """Create a database service."""
 
-        super(DatabaseService, self).__init__(service_name="cloudDatabases",
```
Comment (WalterLenuzza) [inline rackspace/database/database_service.py]: Rackspace's service type `rax:database` is correct but OpenStack's `database` is not being overridden anymore.

**Comment by WalterLenuzza** [comment]
@briancurtin The new `_find` methods are awesome, and the Cloud Backup code seems to work fine.

But Cloud Databases resources are not being loaded: looks like services are not overridden in the profile if their names don't match (makes sense). Any way that we can inherit from `openstack.database` even though the service name is different?

```
>>> from rackspace import connection
>>> import sys
>>> import openstack
>>> openstack.utils.enable_logging(debug=True, stream=sys.stderr)
>>> conn = connection.Connection(username="XXX", api_key="XXX", region="XXX")
2015-11-25 10:00:53,817 DEBUG: openstack.profile Overriding object-store with {'service_type': 'object-store', 'interface': None, 'version': None, 'region_name': None, 'service_name': None}

>>> db_baks = [x for x in conn.database.backups()]
Traceback (most recent call last):
  File "<input>", line 1, in <module>
AttributeError: 'Proxy' object has no attribute 'backups'
>>> conn.database.__module__
'openstack.database.v1._proxy'
```

**Comment by briancurtin** [comment]
Good find - I'll look into it. We don't provide a great way from upstream to override resources downstream like we're doing with databases, so it's a bit manual. At the very least, I can make it work for the time being while we improve it up there.

**Comment by briancurtin** [comment]
If you took out that "import openstack" line, you'd have done what I tested, which works. Ugh, that's not great.

```python
>>> from rackspace import connection
>>> conn = connection.Connection(username="", api_key="", region="")
>>> conn.database
<rackspace.database.v1._proxy.Proxy object at 0x10dc4ef98>
>>> conn.database.__module__
'rackspace.database.v1._proxy'
>>> list(conn.database.backups())
[] # I don't have any of these, but at leas the call worked properly.
```

**Comment by briancurtin** [comment]
Ok, this is more odd. `import openstack` by itself usually does nothing, and you would not be able to call `openstack.utils.enable_logging` as none of that is plumbed all the way through to be supported off of importing just the top level namespace. However, if you do that after already having done `from rackspace import connection`, what you did works, but only in that case. Whatever is allowing that to work seems to also be messing up the service issue.

**Comment by briancurtin** [comment]
Alright, well this is weird, but not entirely weird, but also avoidable. At least for now. If you change your import to `from openstack import utils` and then adjust accordingly, it works as expected.

```python
>>> from rackspace import connection
>>> from openstack import utils
>>> utils.enable_logging(True, path="sdk.log")
>>> conn = connection.Connection(username="", api_key="", region="")
>>> list(conn.database.instances())
[] # again this is working, i just have no instances
>>> conn.database
<rackspace.database.v1._proxy.Proxy object at 0x104e96f60>
```

The `import openstack` line by itself only works because under the hood, most of what it contains has already been loaded in the `from rackspace import connection` line. I don't know exactly why it's stomping on the database service that gets loaded, but for right now I'm going to say that the way you did it is "not supported". None of this is setup to work when imported only as a top-level name.

Additionally, for fun, if you did `import openstack` first and then `from rackspace import connection` second, nothing on the `openstack` name would work until the second line has been interpreted, but after that, everything would work as we need because the proper `rackspace.database` gets loaded and doesn't end up being stomped on.

**Comment by WalterLenuzza** [comment]
Can't say I understand why this happened, but I've noticed the weirdness before. I'll carry on testing my code how you suggested. Would integration tests be the right place to verify that all plugins are loaded properly?

:+1: for merging this

**Pull request merged**